File: 0009_is_palindrome/is_palindrome.go
package main

func isPalindrome(x int) bool {
	rev := 0
	if x < 0 || (x != 0 && x%10 == 0) {
		return false
	}
	for x > rev {
		rev = (rev * 10) + x%10
		x /= 10
	}
	return x == rev || x == (rev/10)
}

func main() {

}

File: 0234_palindrome_linked_list/is_palindrome.py
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None


def is_palindrome(head: ListNode) -> bool:
    """
    Time: O(n)
    Space: O(n)
    """
    stack = []
    t = h = head
    while h and h.next:
        stack.append(t.val)
        h = h.next.next
        t = t.next
    if h:
        t = t.next
    while t:
        if t.val != stack.pop():
            return False
        t = t.next
    return True
